skbio/metadata/_interval.py

```diff
@@ -90,6 +91,10 @@ def __init__(self, interval_metadata, locations,
 
     def _add(self):
         """Add the current ``Interval`` to the IntervalMetadata object."""
+        if self.locations[-1][-1] > self._interval_metadata.upper_bound:
```

Comment:
Should we have an explicit `lower_bound` that is always zero, then validate that here? It seems that Sequence objects require their interval metadata to be bounded by `[0, upper_bound)`, where `upper_bound` is `len(seq)`.